ticdc/ticdc-architecture.md

```diff
@@ -0,0 +1,180 @@
+    - global ResolvedTS: The minimum ResolvedTS of all Processors on all TiCDC nodes. Because each TiCDC node has one or more Processors, each Processor corresponds to multiple table pipelines.
+
+    For TiCDC, the ResolvedTS sent by TiKV is a special event in the format of `<resolvedTS: timestamp>`. In general, the ResolvedTS satisfies the following constraints:
+
+    ```
+    table resolved TS >= local resolved TS >= global Resolved TS
+    ```
+
+#### CheckpointTS
+
+This timestamp only exists in TiCDC. It indicates the minimum data changes that have been replicated to the downstream. That is, for TiCDC, the data before this timestamp has been replicated to the downstream system.
+
+- Table checkpointTS: Because TiCDC replicates data in tables, the table checkpointTS indicates the minimum data that has been replicated at the table level.
+- Processor checkpointTS: Indicates the minimum checkpointTS among all Processors.
+- Global checkpointTS: Indicates the minimum checkpointTS among all Processor checkpointTS.
+
+Generally, a checkpoint TS satisfies the following constraint:
+
+```
+table checkpoint TS >= local checkpoint TS >= global checkpoint TS
+```
+
+The complete constraint is as follows if we also include ResolvedTS:
+
+```
+table resolved TS >= local resolved TS >= global Resolved TS >= table checkpoint TS >= local checkpoint TS >= global checkpoint TS
+```
+
+After data changes and transactions are committed, the resolvedTS on the TiKV node will continue to advance, and the Puller module on the TiCDC node keeps receiving data pushed by TiKV. The Puller module also decides whether to scan incremental data based on the received information, which ensures that all data changes are sent to the TiCDC node.
+
+The Sorter module sorts data received by the Puller module in ascending order according to the timestamp. This process ensures data consistency at the table level. Next, the Mounter module assembles the data changes from the upstream into the format that the Sink module can consume, and sends it to the Sink module. The Sink module replicates the data changes between the checkpointTS and the ResolvedTS to the downstream in the order of the TS, and advancing the checkpointTS.
+
+The preceding sections only cover data changes of DML statements and do not include DDL statements. The following sections introduce the timestamp related to DDL statements.
+
+#### Barrier TS
+
+Barrier TS is generated when a DDL statement is executed or a Syncpoint is used.
+
+- This timestamp ensures that all changes before this DDL statement are replicated to the downstream. After this DDL is executed, data changes are replicated again. Because DDL statements are processed by the Capture Owner, the Barrier TS corresponding to the DDL statement is only generated by the Processor thread of the owner node.
```

Comment:
```suggestion
- This timestamp ensures that all changes before this DDL statement are replicated to the downstream. After this DDL statement is executed and replicated, TiCDC starts replicating other data changes. Because DDL statements are processed by the Capture Owner, the Barrier TS corresponding to a DDL statement is only generated by the Processor thread of the owner node.
```